Sources/Commands/PackageCommands/PluginCommand.swift

```diff
@@ -373,7 +379,7 @@ struct PluginCommand: SwiftCommand {
         // TODO: We should also emit a final line of output regarding the result.
     }
 
-    static func availableCommandPlugins(in graph: ModulesGraph, limitedTo packageIdentity: String?) -> [PluginModule] {
+    static func availableCommandPlugins(in graph: ModulesGraph, limitedTo packageIdentity: String?) -> [ResolvedModule] {
```

Comment:
IIUC we force cast it to `PluginModule` at all call sites, not sure why the return type has changed?

```suggestion
    static func availableCommandPlugins(in graph: ModulesGraph, limitedTo packageIdentity: String?) -> [PluginModule] {
```

Reply:
It's because of `PluginCommand.run` that uses findPlugins to filter thing and takes a `ResolvedModule` but maybe it doesn't actually need a `ResolvedModule`, I'll take a look.

Reply:
We need that for `preparePluginTools` inside of `PluginCommand.run` so yeah, both of these produce `ResolvedModule` to avoid having to re-lookup stuff in the graph.